src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = { ...state };
```

Comment:
NOTE: this is not critical, just a suggestion for a small optimization.

You're making double copying of `state` object - on lines 11 and 14.
Copying is certainly needed on line 14 - because on previous iteration we don't do it (line 31). But on line 11, before the loop, we can just do `let currentState = state`.